linodecli/api_request.py

```diff
@@ -65,6 +96,36 @@ def do_request(
     return result
 
 
+def _merge_results_data(results: Iterable[dict]):
+    """Merge multiple json response into one"""
+
+    iterator = iter(results)
+    merged_result = next(iterator, None)
+    if not merged_result:
+        return None
+
+    if "pages" in merged_result:
+        merged_result["pages"] = 1
+    if "page" in merged_result:
+        merged_result["page"] = 1
+    if "data" in merged_result:
+        merged_result["data"] += list(
+            itertools.chain.from_iterable(r["data"] for r in iterator)
+        )
+    return merged_result
+
+
+def _get_all_pages_results_generator(
```

Comment:
nit: the method name sounds a bit odd. Can we align with the style of other methods to name it like `_get_all_pages_results` or `_generate_all_pages_results`? And add a quick description will be helpful as well.